### PR TITLE
Aditya/appointment options

### PR DESCRIPTION
**Pull request by Aditya-Bhandare**
1. Migrated from activity to fragment for HomeActivity.
2. Updated onCreateOptionMenu for both fragments
3. Resolved conflicts from Master branch

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -13,8 +13,9 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.Covid19VaccineTracker">
         <activity android:name=".ui.center.CenterActivity"></activity>
-        <activity android:name=".MainActivity" />
-        <activity android:name=".ui.HomeActivity">
+        <activity android:name=".MainActivity" android:launchMode="singleTask" />
```
Comment (gaurav7891) [inline app/src/main/AndroidManifest.xml]: Please revert this if we are done with the changes for fragment implementation

File: app/src/main/java/com/nativework/covid19vaccinetracker/base/BaseApp.kt [hunk 2]
```diff
@@ -1,14 +1,58 @@
...
+
+        when (item.itemId) {
+            android.R.id.home -> {
+                onBackPressed()
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+  /*  override fun onBackPressed() {
```
Comment (gaurav7891) [inline app/src/main/java/com/nativework/covid19vaccinetracker/base/BaseApp.kt]: remove the unwanted code

File: app/src/main/java/com/nativework/covid19vaccinetracker/ui/HomeActivity.kt [hunk 3]
```diff
@@ -225,12 +29,15 @@ class HomeActivity : BaseApp(), RecentSearchAdapter.OnClickListener {
...
+
         when (item.itemId) {
             R.id.menu_searchByPin -> {
-                binding.homeContainer.visibility = View.VISIBLE
-                binding.btnSearch.visibility = View.GONE
                 loadFragment(AppointmentFragment.newInstance(), R.id.home_container)
             }
+
+            R.id.menu_searchByDistrict -> {
+                loadFragment(HomeFragment(), R.id.home_container)
```
Comment (gaurav7891) [inline app/src/main/java/com/nativework/covid19vaccinetracker/ui/HomeActivity.kt]: Please change the name with some meaningFul class name, based on responsibilty of the class, HomeFragmemt() and AppointmentFragment doesnt match well

File: app/src/main/java/com/nativework/covid19vaccinetracker/ui/appointment/AppointmentFragment.kt [hunk 4]
```diff
@@ -94,11 +96,11 @@ class AppointmentFragment : BaseFragment() {
         binding.calendarView.minDate = date
     }
 
-    private fun  getPincode(): Int {
-        val pin :String = binding.edtPincode.text.trim().toString()
-        if (!pin.isNullOrEmpty()){
+    private fun getPincode(): Int {
+        val pin: String = binding.edtPincode.text.trim().toString()
+        if (!pin.isNullOrEmpty()) {
```
Comment (gaurav7891) [inline app/src/main/java/com/nativework/covid19vaccinetracker/ui/appointment/AppointmentFragment.kt]: condition can be simplified as pin.isNotEmpty()

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -2,4 +2,8 @@
     <string name="app_name">VaccineTracker</string>
     <!-- TODO: Remove or change this placeholder text -->
     <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (gaurav7891) [inline app/src/main/res/values/strings.xml]: remove this string

**Review by gaurav7891 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/nativework/covid19vaccinetracker/ui/HomeActivity.kt (hunk 3 above)
Comment (Aditya-Bhandare) [reply app/src/main/java/com/nativework/covid19vaccinetracker/ui/HomeActivity.kt]: Changed to CalenderByDistrictFragment() and CalenderByPincodeFragment()

**Review by gaurav7891 — APPROVED** [review]

**Pull request merged**